### ISSUE TITLE
load_balancers and target_group_arns are not always set on aws_autoscaling_group resource

### ISSUE THREAD

**Issue by antonbabenko** [terraform-providers/terraform-provider-aws#2412]
Hi there,

### Terraform Version
```
Terraform v0.11.0
+ provider.aws v1.3.1
```

### Affected Resource(s)
- aws_autoscaling_group

### Expected Behavior
`load_balancers` and `target_group_arns` are not always set on `aws_autoscaling_group` resource and this will fail:

```hcl
output "this_autoscaling_group_load_balancers" {
  value       = ["${element(concat(aws_autoscaling_group.this.*.load_balancers, list("")), 0)}"]
}

output "this_autoscaling_group_target_group_arns" {
  value       = ["${element(concat(aws_autoscaling_group.this.*.target_group_arns, list("")), 0)}"]
}
```

Ref: https://github.com/terraform-aws-modules/terraform-aws-autoscaling/commit/95f08bb29cec286349cece3593b2b0e7c9d5b2ee

### Actual Behavior
```
* module.autoscaling.module.autoscaling_group.output.this_autoscaling_group_load_balancers: Resource 'aws_autoscaling_group.this' does not have attribute 'load_balancers' for variable 'aws_autoscaling_group.this.*.load_balancers'
* module.prod.module.asg.module.autoscaling_group.output.this_autoscaling_group_target_group_arns: Resource 'aws_autoscaling_group.this' does not have attribute 'target_group_arns' for variable 'aws_autoscaling_group.this.target_group_arns'
```

### Steps to Reproduce
1. `terraform apply`

### References
The similar one, but for another resource:
- GH-2392

**Comment by thiagolsfortunato** [comment]
+1

**Comment by bflad** [comment]
Hi @antonbabenko 👋 Hope all is well with you. I marked this to close with #7818 (releasing in version 2.1.0) because that pull request confirms that `load_balancers` and `target_group_arns` should always have their attributes set in the state, even with empty lists. I believe when looking back through the older revisions of the resource, it should have also been doing the same as well.

That said, we have had some frustrating behavior in Terraform 0.11 and earlier where if any of the resource attributes had their `ValidateFunc` fail, Terraform would return the `does not have attribute` error instead of the validation error, e.g. https://github.com/hashicorp/terraform/issues/18129 😖 Good news is that is fixed in Terraform 0.12.

If we are continuing to see odd behavior with `does not have attribute` with Terraform AWS Provider version 2.1.0 or later and (preferably) with Terraform 0.12-beta1 or later, let's triage it in a new GitHub issue with a debug log and we'll get to the bottom of it. 👍

**Comment by antonbabenko** [comment]
Hi Brian! Long time to see IRL :)

Thanks for a great update.

I appreciate such a detailed explanation for the behavior in Terraform 0.11 and I will keep watching if other resources have similar errors. I am slowly updating some of the `terraform-aws-modules` to be Terraform 0.12 friendly.
